Adding WDL and Docker Contribution Guides

tefirman opened the pull request:
As the WILDS WDL pipelines prepare to make their debut in a week and a half, we need guidelines around how WDL's and their corresponding Docker images should be designed. This pull request adds initial versions for each set of guidelines and incorporates them into the overarching WILDS Contribution Guide after the "Package Maintenance" section but before the "Security" section. The location of these guides is definitely flexible, just wanted to get an initial version out there and we can move it as needed in the future. These are definitely first drafts, so if you'd like to add/remove guidelines or you feel that something's misworded/misarranged, please feel free to suggest those changes. Thanks for your input! Really appreciate it! #IntoTheWildsWeGo :wolf:

sitapriyamoorthi commented on wdlconfig.qmd:
This is just my personal opinion but starting of with saying "So as not to reinvent the wheel..." Might give the reviewer the impression that okay so this  WDL pipelines simply follows these two guidelines. My take would be just to be upfront and say something like  the WDL configuration guide is inspired by the BioWDL and WARP guidelines.

tefirman replied:
Totally agree, it starts things off on the wrong foot. Even if we are very similar to the other two, it's best to establish the guidelines as our own. I'll get that updated right now.

sitapriyamoorthi commented on wdlconfig.qmd:
Is this the philosophy of WILD or WDL WILDS?
Alternative sentence idea:
Each WILDS repository is ideally a self-contained demonstration of a specific bioinformatic workflow.

tefirman replied:
Good point, probably best to be more specific in saying "WILDS WDLs" rather than WILDS in general.

sitapriyamoorthi commented on wdlconfig.qmd:
A little confused about these steps. Like are these what we expect the user to do? My take would be that we add a little more information. Something perhaps like this:
1. Each repository will have a Readme file that will detail the over-arching goals of that workflow (for example this workflow will help you convert a fastq file into a cram file). Provide some details about what type of file sources are expected etc. basically everything a user needs to know about the workflow in order to decide if they want to fork and run the workflow. 
2. Once you have decided that this repository is useful to you, you can clone the repository as is, specify your inputs and execute the workflow. 
3. You can also take inspiration from a repository and create a new workflow and then submit it back to the WILDS for review and inclusion.

sitapriyamoorthi commented on wdlconfig.qmd:
My take is not to keep referring back to the BioWDL or WARP recommendations. But just to be very straight about what our rules are and this is what you should expect.

sitapriyamoorthi commented on wdlconfig.qmd:
Maybe add a section on what the user should expect. For example... when you enter a WILDS WDL repo you will find:
1. A readme
2. A WDL
3. An example JSON 
4. Example/test inputs (optional)

sitapriyamoorthi reviewed (changes requested):
Hey I think it would just be easier for me to pull a branch make recommendations and then send them over.

tefirman commented:
@sitapriyamoorthi -- I pushed some updates to address your comments on the WDL Config file, would you mind taking a look one more time?

laderast commented on docker.qmd:
```diff
@@ -0,0 +1,49 @@
+
+## Repository Guidelines
+
+- In terms of the repo organization, each image should have its own directory named after the tool being used in the image. Each version of the image should have its own Dockerfile in that directory following the naming convention of `[IMAGENAME]/Dockerfile_[VERSIONTAG]`.
```
This is a lot more work, but I think that having examples might be really helpful to some of the more abstract guidelines. At least with relevant lines, and you can point towards code examples in getwilds.

tefirman replied:
Great call, I think links to code examples in getWILDS would be perfect, especially for the WDL guidelines.

tefirman commented:
Great call, @caalo, thanks for catching that! Just added a few bulletpoints to the Stylistic Guidelines section to address that and added a couple more links out to the WDL101 course in case people want to learn more.
